common/test-date-provider.ts

```diff
@@ -0,0 +1,56 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {WithLogger} from './logger';
+
+/**
+ * A test implementation of a Date-like object.  Used for testing purposes.  It provides a
+ * getTime method that returns a pre-determined fake date value, allowing for
+ * deterministic testing of time-dependent functionality.
+ */
+class TestDateLike {
```

Comment:
Why do we have these test files in `common/`?

Reply:
Mainly because they are going to be used by both system-tests and tests. At another time I tried importing a file in the system tests from the test folder and it ran both test suites when I ran `npm run system-test`. So these shared files have to live somewhere other than `test` or `system-test`.